### PR TITLE
Upgrade kube-state-metrics chart

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Upgrade `kube-state-metrics` chart to the v2.9.4 of the upstream
- Make use of the custom `selectorLabels` template

**Related issue(s)**
See also #9897

### CODE REVIEW

File: resources/monitoring/charts/kube-state-metrics/Chart.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ keywords:
 - monitoring
 - prometheus
 - kubernetes
-version: 2.8.2
+version: 2.9.4
+appVersion: 1.9.7
```
Comment (suleymanakbas91) [inline resources/monitoring/charts/kube-state-metrics/Chart.yaml:10]: Please delete `appVersion` field as discussed.

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: resources/monitoring/charts/kube-state-metrics/Chart.yaml (hunk 1 above)
Comment (skhalash) [reply resources/monitoring/charts/kube-state-metrics/Chart.yaml]: Done

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**